Migrate content from `meta/` to the team compass

choldgraf opened the pull request:
This PR migrates internal team documentation from `meta/` to the team compass. It tries to pull in any information that is "more for the team than it is for the external world". It'll be paired with a PR that removes this content from `meta/` as well.

### References

- closes https://github.com/executablebooks/meta/pull/770 by superceding it

gregcaporaso reviewed (approved):
This looks good to me @choldgraf, thank you!

choldgraf commented on docs/about/history.md:
```diff
@@ -2,8 +2,10 @@
+The Executable Books project began as a grant-funded collaboration between groups at [The Australian National University](https://anu.edu.au),
+[Northern Arizona University](https://nau.edu/), and
+[The University of California at Berkeley](https://www.berkeley.edu/).
+These teams represented many open source projects in the scientific community, such as [QuantEcon](https://quantecon.org), [QIIME](https://qiime2.org/), and [The Jupyter Project](https://jupyter.org/).
```
```suggestion
These teams represented many open source projects in the scientific community, such as [QuantEcon](https://quantecon.org), [QIIME 2](https://qiime2.org/), and [The Jupyter Project](https://jupyter.org/).
```

choldgraf commented on docs/about/team.md:
```diff
@@ -1,5 +1,8 @@
+(team)=
+# Team members
 
+This project has had contribution from many different people and organizations.
```
```suggestion
This project has had contributions from many different people and organizations.
```

choldgraf commented:
I've addressed your suggestions @gregcaporaso - also made some minor naming changes and updates to avoid confusion.

- The `conventions.md` file now `{include}`s the file at the `.github/` repository and we have an explanation for how this works to make it easier to follow.

choldgraf commented:
I'm gonna merge this one in since we've already got a +1 from @gregcaporaso and can iterate from there
